launchdarkly-android-client-sdk/src/main/java/com/launchdarkly/sdk/android/ClientContextImpl.java

```diff
@@ -123,6 +123,21 @@ public static ClientContextImpl forDataSource(
         );
     }
 
+    /**
+     * Sets the evaluation context and returns a new instance of {@link ClientContextImpl}
+     * @param context to now use as the evaluation context
+     * @return a new instance
+     */
+    public ClientContextImpl setEvaluationContext(LDContext context) {
+        return new ClientContextImpl(
+            super.setEvaluationContext(context),
+            this.diagnosticStore,
+            this.fetcher,
+            this.platformState,
+            this.taskExecutor
+        );
+    }
```

Comment:
For reviewers:  LDClient's ClientContextImpl member variable should be the authority on the current context.  This allows us to change the context in that object so that can be true after an identify call.

Reply:
I have some concerns with this. Copies of the client context are passed to various components, and those client contexts do not have a centralized update mechanism. So putting the evaluation context into the client context provides easy access to a stale or unset LDContext.

Reply:
You are correct.  I believe that is how it has been.  Any components that use the context have an alternative mechanism for updating it when it changes.